cloud/passportauth/integration/UserApi.ts

```diff
@@ -0,0 +1,40 @@
+import {BaseUser} from '../src/index';
+import {DataRepository} from './UserDataRepository';
+import {User} from './UserSeedData';
+
+/**
+ * A sample implementation of a base user api
+ */
+export class UserApi implements BaseUser {
```

Comment:
As we agreed before. This entire class is not really needed as it serve no purpose. 
We can:
- Have BaseUser interface with set of requirements as methods:
```
user.getId
user.getPassword
user.getLogin
user.getRole
```

In this particular example:
getId implementation will be:

`return this.myCustomId`

Interface is implemented by *module integrators* - **mapping**

- Have single method in repository to get user object. 
```
getUserByLogin
```

I have no strong feelings to have this but you may find this really inflexible later when implementing actual solution. 

This is implemented by module *integrators* **Data Source**